### PR TITLE
feat(common-lib): derive dynamic field instead of call to fullnode 

### PR DESCRIPTION
**Pull request by Tzal3x**
- Upgrade @mysten/sui to 1.12.0
This version includes the `deriveDynamicFieldID`
function needed to calculate the DF object ID
without making a request to a fullnode.
- Replace `client.getDynamicFieldObject` with `deriveDynamicFieldID`.
- Skip `page_fetching` benchmarks. The results we get from it are misleading and slow down development whenever there are changes happening in its' dependencies. In the near future we should revisit how to properly bench this. https://github.com/MystenLabs/walrus-sites/issues/250

TODO (following PR)
- ⚡ Performance improvement: Use `multiGetObjects` to fetch both the display and the dynamic field object in the same request.

### CODE REVIEW

File: portal/common/lib/resource.test.ts [hunk 1]
```diff
@@ -190,53 +182,11 @@ describe('fetchResource', () => {
...
+        getObject.mockResolvedValueOnce(undefined);
 
-        const result = await fetchResource(mockClient, '0xParentId', '/path', seenResources);
+        const result = await fetchResource(mockClient, '0x1', '/path', seenResources);
 
         // Check that the function returns NOT_FOUND
         expect(result).toBe(HttpStatusCodes.NOT_FOUND);
     });
-
-    test('should correctly handle a chain of redirects', async () => {
```
Comment (giac-mysten) [inline portal/common/lib/resource.test.ts:201]: question: has this been removed?

File: portal/common/lib/resource.ts [hunk 2]
```diff
@@ -41,38 +47,25 @@ export async function fetchResource(
...
     }
 
-    // Initiate a pre-fetch for the checkRedirect operation without resolving it.
-    // We don't need the result yet, but it's useful if we will need
-    // it later, so we don't have to loose time.
-    const checkRedirectPromise = checkRedirect(client, objectId);
+    let redirectId  = await checkRedirect(client, objectId);
+    if (redirectId) {
+        fetchResource(client, redirectId, path, seenResources, depth + 1)
+    }
```
Comment (giac-mysten) [inline portal/common/lib/resource.ts:53]: Question/important: i think we still want to do the background prefetch of the redirect, so that we first check if there is a resource, and only if there isn't we redirect. Why the change?

File: portal/common/lib/resource.ts [hunk 3]
```diff
@@ -41,38 +47,25 @@ export async function fetchResource(
...
     const pageData = await client.getObject({
-        id: dynamicFields.data.objectId,
+        id: dynamicFieldId,
         options: { showBcs: true },
     });
 
     // If no page data found.
-    if (!pageData.data) {
+    if (!pageData || !( pageData.data )) {
         console.log("No page data found");
```
Comment (giac-mysten) [inline portal/common/lib/resource.ts:69]: Suggestion: the `await` for the redirect could be here?

**Review by giac-mysten — APPROVED** [review]
Awesome, thanks @Tzal3x . Generally looks good, only one more important comment.

File: portal/common/lib/resource.test.ts (hunk 1 above)
Comment (Tzal3x) [reply portal/common/lib/resource.test.ts:201]: Yes, I found out it was redundant.
